dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java

```diff
@@ -227,8 +226,16 @@ public static void runCLI(Context context, DOIOrganiser organiser, String[] args
                 }
 
                 for (DOI doi : dois) {
-                    organiser.reserve(doi);
-                    context.uncacheEntity(doi);
+                    doi = context.reloadEntity(doi);
```

Comment:
The `reloadEntity` has been introduced to reload `detached` entities after a `context.rollback()`?

Reply:
Yes, rollback or commit.  Both cause database entities to be detached.